attest/tpm_linux.go

```diff
@@ -359,48 +358,40 @@ func allPCRs12(ctx *tspi.Context) (map[uint32][]byte, error) {
 }
 
 // PCRs returns the present value of all Platform Configuration Registers.
-func (t *TPM) PCRs() (map[int]PCR, tpm2.Algorithm, error) {
+func (t *TPM) PCRs(alg HashAlg) ([]PCR, error) {
```

Comment:
nit: godoc needs to be updated.

It's also kinda weird that we're defining public APIs in _linux.go and _windows.go files. Maybe add a TODO for something similar to key_{linux,windows}.go where the public API lives in a file without build tags, and the OS specific files only contain private methods?